streampark-console/streampark-console-service/src/main/assembly/conf/config.yaml

```diff
@@ -35,10 +35,11 @@ server:
 # system database, default h2, mysql|pgsql|h2
 datasource:
   dialect: h2  #h2, mysql, pgsql
+  h2-data-dir:  # if datasource.dialect is h2 ,you can configure the data dir
```

Comment:
```suggestion
  h2-data-dir:  /tmp/streampark/h2-data # if datasource.dialect is h2 ,you can configure the data dir
```

I think it's better to set a default value for it. And it should be the same as default value.

Reply:
Users may not necessarily have permission for this directory. So I think it is not suitable to provide a default value.

Reply:
On Linux systems, all users have permissions on the `/tmp` directory by default.

Reply:
> On Linux systems, all users have permissions on the `/tmp` directory by default since it's default permissions is `777`.

"/tmp" is a system temporary directory. I am concerned that it might get cleaned, as some users have scripts that regularly clean up directories and files in "/tmp." Therefore, it is not recommended to place it in "/tmp." It is better for users to manually specify the path using "h2dir" or have the program directly locate the directory under the user's home directory. What do you think？

Reply:
I don't think leaving the default value blank is a good option since we using `h2` as default database. Providing the default value allows users to use it directly without the need to configure it. If users use it in a production environment, we can suggest that users change to other values in the documentation. WDYT?

Reply:
> I don't think leaving the default value blank is a good option since we using `h2` as default database. Providing the default value allows users to use it directly without the need to configure it. If users use it in a production environment, we can suggest that users change to other values in the documentation. WDYT?

We agree with your opinion that default values should be set, but "/tmp/xxx" is not a good solution. Perhaps "~/streampark/metadata" is a good idea, using "~" will automatically locate the user directory.

Reply:
If it is for demonstration purposes, I prefer the /tmp directory. However, we need to consider users who might use it to run lightweight jobs and need persistence. In this case, the /tmp directory might not be suitable. I prefer the /data/streampark/ directory. This not only avoids the habitual cleaning of the /tmp directory that might lead to data loss but also aligns with the conventions of some projects, such as:

Docker: By default, Docker containers use /data for storing persistent data.
Elasticsearch: The default data directory for Elasticsearch is /var/lib/elasticsearch which often gets customized to /data/elasticsearch.
MongoDB: Uses /data/db as the default directory for storing data.
Prometheus: The default data directory is /var/lib/prometheus, and many setups prefer using /data/prometheus.
These examples demonstrate a common practice of using the /data directory for persistent storage, ensuring stability and avoiding data loss due to temporary directory cleaning.

What do you think?

Reply:
> > I don't think leaving the default value blank is a good option since we using `h2` as default database. Providing the default value allows users to use it directly without the need to configure it. If users use it in a production environment, we can suggest that users change to other values in the documentation. WDYT?
> 
> We agree with your opinion that default values should be set, but "/tmp/xxx" is not a good solution. Perhaps `~/streampark/metadata` is a good idea, using `~` will automatically locate the user directory.

Using the ~ directory might lead to data loss when users change or leave and their accounts get deleted. Of course, we can also inform users about this potential issue in the documentation.

Reply:
> We agree with your opinion that default values should be set, but "/tmp/xxx" is not a good solution. Perhaps `~/streampark/metadata` is a good idea, using `~` will automatically locate the user directory.

Using `~/streampark/metadata` is good to me.

Reply:
> > We agree with your opinion that default values should be set, but "/tmp/xxx" is not a good solution. Perhaps `~/streampark/metadata` is a good idea, using `~` will automatically locate the user directory.
> 
> Using `~/streampark/metadata` is good to me.

Thanks, updated~